### PR TITLE
fix compatibility issues with llvm 15

### PR DESCRIPTION
**Pull request by vincentAlba**
VectorType can't access GetNumElements without being cast to FixedVectorType since llvm 13
For now the code raise an error if Scalable vector are used

The following test also fail in the current llvm version :
8,9,12,19,34

### CODE REVIEW

**Comment by pablooliveira** [comment]
Thanks @vincentAlba ! Nice work.

Before merging, we will need to look into fixes for the failing tests.
Also the linter is failing some tests; could you please fix the formatting of your patch by running clang-format (default arguments) on the .c / .cpp files you changed ?

**Comment by pablooliveira** [comment]
I see that the command ` -enable-new-pm=0 ` which is required for retro-compatibility in new LLVMs is not supported in LLVM 7. I think we should add this option conditionally depending on the LLVM version.

File: verificarlo.in.in [hunk 1]
```diff
@@ -223,18 +229,30 @@ def compiler_mode(sources, options, output, args):
         if args.inst_fcmp:
             extra_args += "-vfclibinst-inst-fcmp "
 
+        llvm_version = get_llvm_version()
+
         if args.inst_func:
             # Apply function's instrumentation pass
-            shell(
-                f'{opt} -S -load {libvfcfuncinstrument} -vfclibfunc {ir.name} -o {ins.name}')
+            if llvm_version >= 15:
```
Comment (titeup) [inline verificarlo.in.in:236]: new pass manger start at 13. I tested your code with LLVM 13 and it works fine.

File: verificarlo.in.in [hunk 2]
```diff
@@ -223,18 +229,30 @@ def compiler_mode(sources, options, output, args):
...
+                    f'{opt} -S -load {libvfcfuncinstrument} -vfclibfunc {ir.name} -o {ins.name}')
             ir = ins
             ins = get_tmp_filename(basename, '.3.ll', args)
 
         # Apply MCA instrumentation pass
-        shell((f'{opt} -S  -load {libvfcinstrument} '
-               f' -vfclibinst-vfcwrapper-file {vfcwrapper_ir.name} '
-               f' -vfclibinst {extra_args} {selectfunction} '
-               f' {ir.name} -o {ins.name}'))
+        if llvm_version >= 15:
```
Comment (titeup) [inline verificarlo.in.in:246]: new pass manger start at 13. I tested your code with LLVM 13 and it works fine.

**Review by titeup — CHANGES_REQUESTED** [review]
Please change the test to version 13 (and merge and rebase with the latest master), otherwise it looks good.
Thanks for your useful contribution!

**Comment by pablooliveira** [comment]
@vincentAlba all tests are now passing. The failing tests were unrelated to your patch (#305) and have been fixed.

If you implement the two changes requested: 

- activate the `enable-new-pm` option conditionally depending on the LLVM version.

- change the llvm_version test to 13 as suggested by @titeup 

we can merge this :tada:

**Review by pablooliveira — CHANGES_REQUESTED** [review]

File: verificarlo.in.in [hunk 3]
```diff
@@ -225,16 +226,26 @@ def compiler_mode(sources, options, output, args):
...
+                    f'{opt} -S -load {libvfcfuncinstrument} -vfclibfunc {ir.name} -o {ins.name}')
             ir = ins
             ins = get_tmp_filename(basename, '.3.ll', args)
 
         # Apply MCA instrumentation pass
-        shell((f'{opt} -S  -load {libvfcinstrument} '
-               f' -vfclibinst-vfcwrapper-file {vfcwrapper_ir.name} '
-               f' -vfclibinst {extra_args} {selectfunction} '
-               f' {ir.name} -o {ins.name}'))
+        if int(llvm_version) >= 13:
```
Comment (pablooliveira) [inline verificarlo.in.in:239]:
```suggestion
        # For LLVM >= 13 we fallback to the legacy pass manager
        if int(llvm_version) >= 13:
```

**Review by pablooliveira — APPROVED** [review]

**Comment by pablooliveira** [comment]
This looks good @vincentAlba, merging, thanks for the contribution.
Unrelated but could you please open a PR with the new usercall to change precision in different parts of the code we discussed together ? :-)

**Pull request merged**